### PR TITLE
new package: py-oauthlib

### PR DESCRIPTION
**Pull request by Sinan81**
missing dependencies (blinker, pyjwt) will be pushed soon:

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-oauthlib/package.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    url      = "https://pypi.io/packages/source/o/oauthlib/oauthlib-3.1.0.tar.gz"
+
+    version('3.1.0', sha256='bee41cc35fcca6e988463cacc3bcb8a96224f470ca547e697b604cc697b2f889')
+    version('2.0.2', sha256='b3b9b47f2a263fe249b5b48c4e25a5bce882ff20a0ac34d553ce43cff55b53ac')
+
+    depends_on('py-setuptools', type='build')
+
+    depends_on('py-pyjwt@1.0.0:', type=('build', 'run'))
+    depends_on('py-blinker', type=('build', 'run'))
+    depends_on('py-cryptography', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-oauthlib/package.py]: It looks like these are optional. I would add a variant.

File: var/spack/repos/builtin/packages/py-oauthlib/package.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    depends_on('py-mock@2.0:', type='test')
+    depends_on('py-pytest@4.0:', type='test')
+    depends_on('py-pytest-cov@2.6:', type='test')
+
+    depends_on('py-nose', type='test', when='@2.0.2')
+    depends_on('py-unittest2', type='test', when='^python@2 @2.0.2')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-oauthlib/package.py]: Where did you find these test dependencies?

File: var/spack/repos/builtin/packages/py-oauthlib/package.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    conflicts('python@3.1:3.3')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-oauthlib/package.py]:
```suggestion
    depends_on('python@2.7:2.8,3.4:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-oauthlib/package.py (hunk 2 above)
Comment (Sinan81) [reply var/spack/repos/builtin/packages/py-oauthlib/package.py]: these dependencies were listed in setup file for v2.0.2

**Pull request merged**